src/binding.ts

```diff
@@ -0,0 +1,36 @@
+/* --------------------------------------------------------------------------------------------
+ * SonarLint for VisualStudio Code
+ * Copyright (C) 2017-2022 SonarSource SA
+ * sonarlint@sonarsource.com
+ * Licensed under the LGPLv3 License. See LICENSE.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import * as VSCode from 'vscode';
+
+const SONARLINT_CATEGORY = 'sonarlint';
+const PROJECT_BINDING_PROPERTY = 'connectedMode.project';
+
+export class BindingService {
+
+  private static _instance: BindingService;
+
+  static init(): void {
+    BindingService._instance = new BindingService();
+  }
+
+  static get instance(): BindingService {
+    return BindingService._instance;
+  }
+
+  async updateWorkspaceFolderBinding(workspaceFolderUri: VSCode.Uri, projectKey: string): Promise<void> {
```

Comment:
IMHO, this method should also accept an optional connection ID. With this signature, folders can only be bound to a project from the `<default>` connection.

I believe this module should export an interface (e.g `ProjectBinding`) to represent bindings. This method could then take one such object as parameter (and an eventual `getAllBindings` method could return a collection of those).